core/src/main/scala/chisel3/Mem.scala

```diff
@@ -60,42 +61,67 @@ sealed abstract class MemBase[T <: Data](val t: T, val length: BigInt) extends H
     */
   def apply(x: Int): T = macro SourceInfoTransform.xArg
 
+  protected def do_apply_impl(idx: UInt, clock: Clock, dir: MemPortDirection, warn: Boolean)(implicit sourceInfo: SourceInfo, compileOptions: CompileOptions): T = {
+    if (warn && clock != clockInst) {
+      Builder.warning("The clock used to initialize the memory is different than the one used to initialize the port. " + 
+        "If this is intentional, please pass the clock explicitly when creating the port. This behavior will be an error in 3.6.0")
+    }
+    makePort(sourceInfo, idx, dir, clock)
+  }
+
+  /** Creates a read/write accessor into the memory with dynamic addressing.
+    * See the class documentation of the memory for more detailed information.
+    */
+  def apply(x: UInt): T = macro SourceInfoTransform.xArg
+  def apply(x: UInt, y: Clock): T = macro SourceInfoTransform.xyArg
```

Comment:
I don't think I phrased this as precisely as I meant to. Can you make sure each `def <method>` is immediately above the corresponding `def do_<method>` (with only an empty line between them)?

More like:
```scala
  def apply(x: UInt): T = macro SourceInfoTransform.xArg

  def do_apply(idx: UInt)(implicit sourceInfo: SourceInfo, compileOptions: CompileOptions): T = ...

   def apply(x: UInt, y: Clock): T = macro SourceInfoTransform.xyArg

  def do_apply(idx: UInt, clock: Clock)(implicit sourceInfo: SourceInfo, compileOptions: CompileOptions): T = ...
```

Does that make sense?

Reply:
Oh i see what you mean